src/read.rs

```diff
@@ -854,11 +856,70 @@ pub(crate) fn central_header_to_zip_file<R: Read + Seek>(
     let central_header_start = reader.stream_position()?;
 
     // Parse central header
-    let signature = reader.read_u32_le()?;
-    if signature != spec::CENTRAL_DIRECTORY_HEADER_SIGNATURE {
-        Err(ZipError::InvalidArchive("Invalid Central Directory header"))
-    } else {
-        central_header_to_zip_file_inner(reader, archive_offset, central_header_start)
+    let block = ZipEntryBlock::parse(reader)?;
+    central_header_to_zip_file_inner(reader, archive_offset, central_header_start, block)
+}
+
+#[derive(Copy, Clone, Debug)]
+#[repr(packed)]
+pub(crate) struct ZipEntryBlock {
+    pub version_made_by: u16,
+    pub version_to_extract: u16,
+    pub flags: u16,
+    pub compression_method: u16,
+    pub last_mod_time: u16,
+    pub last_mod_date: u16,
+    pub crc32: u32,
+    pub compressed_size: u32,
+    pub uncompressed_size: u32,
+    pub file_name_length: u16,
+    pub extra_field_length: u16,
+    pub file_comment_length: u16,
+    pub disk_number: u16,
+    pub internal_file_attributes: u16,
+    pub external_file_attributes: u32,
+    pub offset: u32,
+}
+
+impl ZipEntryBlock {
+    const SIG_LEN: usize = mem::size_of::<u32>();
```

Comment:
Instead of these constants, can we have a second struct type for signature+block, and call size_of on that directly where needed?

Reply:
Will do! The first time I implemented this, I didn't separate the signature + block (as you've described). I agree that separating it is more confusing--I'll fix this.